apps/server/src/modules/provisioning/service/tsp-provisioning.service.ts

```diff
@@ -0,0 +1,167 @@
+import { AccountSave, AccountService } from '@modules/account';
+import { ClassFactory, ClassService, ClassSourceOptions } from '@modules/class';
+import { RoleService } from '@modules/role';
+import { Injectable } from '@nestjs/common';
+import { NotFoundLoggableException } from '@shared/common/loggable-exception';
+import { RoleReference, UserDO } from '@shared/domain/domainobject';
+import { RoleName } from '@shared/domain/interface';
+import { School, SchoolService } from '@src/modules/school';
+import { UserService } from '@src/modules/user';
+import { ExternalClassDto, ExternalSchoolDto, ExternalUserDto, OauthDataDto, ProvisioningSystemDto } from '../dto';
+import { BadDataLoggableException } from '../strategy/loggable';
+
+@Injectable()
+export class TspProvisioningService {
+	private ENTITY_SOURCE = 'tsp'; // used as source attribute in created users and classes
+
+	constructor(
+		private readonly schoolService: SchoolService,
+		private readonly classService: ClassService,
+		private readonly roleService: RoleService,
+		private readonly userService: UserService,
+		private readonly accountService: AccountService
+	) {}
+
+	public async findSchoolOrFail(system: ProvisioningSystemDto, school: ExternalSchoolDto): Promise<School> {

```

Comment:
This sounds like a function that should rather exists in the SchoolModule, but I won't insist on it now